Libs/MRML/Core/vtkMRMLMeasurement.cxx

```diff
@@ -353,10 +406,25 @@ const char* vtkMRMLMeasurement::GetLastComputationResultAsPrintableString()
 }
 
 //----------------------------------------------------------------------------
-void vtkMRMLMeasurement::SetValue(double value)
+void vtkMRMLMeasurement::SetDisplayValue(double displayValue, const char* units)
 {
-  vtkDebugMacro(<< this->GetClassName() << " (" << this << "): setting Value to " << value);
+  vtkDebugMacro(<< this->GetClassName() << " (" << this << "): setting Value to " << displayValue);
   bool modified = false;
+  if (units)
+    {
+    if (this->Units != units)
+      {
+      this->Units = units;
+      modified = true;
+      }
+    double displayCoefficient = 1.0;
+    if (this->DisplayCoefficient != displayCoefficient)
```

Comment:
This is I think is a bit hard to read, i.e. this if looks like `displayCoefficient` should come as argument that is set. Instead I'd rename the local variable to `defaultDisplayCoefficient`, or even make a const member with this name, or just use the 1.0 scalar in the if.

Reply:
Good point. I've updated the code to let the user specify the display coefficient along with the unit and use the current display coefficient as default.